### ISSUE TITLE
Not able to fire click event on date picker cell

### ISSUE THREAD

**Issue by soru95** [t1m0n/air-datepicker#484]
Libraries used - React.js, Jest, and react-testing-library.

I am writing a wrapper on top of Air-datepicker in React.js for our component library. I have chosen to pass/override 'onSelect' hook while passing the options to Datepicker while instantiating.

While writing the unit test using Jest and react-testing-library, I am not able to invoke onSelect handler through the event mechanism. The steps I am following:

rendered the component.
picked the text input component (a React component) using screen API.
Fired focusIn event on text input, so that date picker will be displayed.
Next, I am trying to select datepicker-cell.
Next, I am trying to fire the click event on datepicker-cell.
If I put the breakpoint in onSelect hook, the control never comes to this hook.
Could you please help, how can I solve this issue?

Test case code:

Approach1:

```javascript
    render(
      <ThemeProvider theme={theme}>
        <SaganDatePicker {...props} />                     // my wrapper React component on top of Air datepicker
      </ThemeProvider>,
     );
    const defaultTextInput = screen.getByLabelText(mainNLS.dateInputLabel);
    expect(defaultTextInput).toBeDefined();
    fireEvent.focus(defaultTextInput);

    const dateCell = screen.getByText(getDateFragment(DATE_FRAGMENTS.DATE));
    expect(dateCell).toBeDefined();
	fireEvent.click(dateCell);
    expect(defaultTextInput.value).toBe(
      getDateFragment(DATE_FRAGMENTS.FULL_DATE),
    );
```

Approach2:

```javascript
    render(
      <ThemeProvider theme={theme}>
        <SaganDatePicker {...props} />                   // my wrapper React component on top of Air datepicker
      </ThemeProvider>,
     );
    const defaultTextInput = screen.getByLabelText(mainNLS.dateInputLabel);
    expect(defaultTextInput).toBeDefined();
    fireEvent.focus(defaultTextInput);
    const datePickerBody = document.body.querySelector(
      '.air-datepicker-body',
    );
    expect(datePickerBody).toBeDefined();
    fireEvent.click(datePickerBody);
    expect(defaultTextInput.value).toBe(
      getDateFragment(DATE_FRAGMENTS.FULL_DATE),
    );
  });
```

What am I doing wrong here? Did I miss anything?

**Comment by soru95** [comment]
@t1m0n Could you please help me in this issue?

**Comment by t1m0n** [comment]
@soru95 Hm, hard to say. Maybe datepicker is not visible? Try to set `{visible: true}` in options.

**Comment by soru95** [comment]
Thanks, @t1m0n for the quick reply.

I tried it quickly, but no help... Actually, in the case of `Today` button, I can fire the event the same way I mentioned above and it's working. Even in the case of months view, clicking on the cell is working fine. The issue is happening with day cells only.

Code of another test case which is working fine (click on month cell):

```javascript
  it(' should add \'Today\' button when view changes to days', () => {
    // months view on initial render
    mockProps.view = 'months';
    render(
      <ThemeProvider theme={theme}>
        <SaganDatePicker {...mockProps} />                   // my wrapper React component on top of Air datepicker
      </ThemeProvider>,
     );
    const defaultTextInput = screen.getByLabelText(mainNLS.dateInputLabel);
    expect(defaultTextInput).toBeDefined();
    fireEvent.focus(defaultTextInput);
    let dpTodayButton = screen.queryByRole('button', { name: /Today/i });
    // no today button in months view
    expect(dpTodayButton).toBeNull();
    const monthCell = screen.getByText(
      getDateFragment(DATE_FRAGMENTS.MONTH_SHORT),
    );
    fireEvent.click(monthCell);
    dpTodayButton = screen.queryByRole('button', { name: /Today/i });
    expect(dpTodayButton.textContent).toBe(mainNLS.datePickerLocale.today);
    hideDatePicker(defaultTextInput);
  });
```


Background:
The requirement is a little different than the original Air-datepicker. So I've added my own `Today` button. which only appears in case of days view and on click of which it selects the current date.

**Comment by t1m0n** [comment]
1. In your case you should trigger event directly on cell. By the way you could get any cell with `getCell` method, it is not described on documentation page, but I'll add it later.  `dp.getCell('2022-02-18').click()` - but datepicker should be visible
2. Default event should work fine.

**Comment by soru95** [comment]
Actually, I found the issue and fixed it. After spending more time on the issue, I found that the issue was happening due to async behavior in `_onChangeSelectedDate`, which was causing the test to get failed due to unmatched timers.

I did small changes in my test and it worked like a charm:

```javascript
      act(() => {
        fireEvent.click(dateCell);
      });
      act(() => {
        jest.advanceTimersByTime(100);
      });
      expect(mockProps.handleSelectDate).toHaveBeenCalledTimes(1);
```

This is just your FYI and for any other person, who might face similar issues in the future.
Thanks very much for all your help and timely responses. Happy coding! :-)
